ouroboros-consensus-cardano/src/cardano-testlib/Test/Consensus/Cardano/ProtocolInfo.hs

```diff
@@ -0,0 +1,361 @@
+  -> HardForkSpec
+  -> ProtocolInfo (CardanoBlock c)
+mkSimpleTestProtocolInfo
+    decentralizationParam
+    securityParam
+    byronSlotLenghtInSeconds
+    shelleySlotLengthInSeconds
+    hardForkSpec
+  = fst
+  $ mkTestProtocolInfo @IO
+      (CoreNodeId 0, coreNodeShelley)
+      shelleyGenesis
+      byronProtocolVersion
+      SL.NeutralNonce
+      genesisByron
+      generatedSecretsByron
+      (Just $ PBftSignatureThreshold 1)
+      hardForkSpec
+  where
+    byronProtocolVersion =
+        CC.Update.ProtocolVersion 0 0 0
+
+    coreNodeShelley = runGen initSeed $ Shelley.genCoreNode initialKESPeriod
+      where
+        initSeed :: Seed
+        initSeed = Seed 0
+
+        initialKESPeriod :: SL.KESPeriod
+        initialKESPeriod = SL.KESPeriod 0
+
+    pbftParams :: PBftParams
+    pbftParams = Byron.byronPBftParams securityParam (NumCoreNodes 1)
+
+    generatedSecretsByron :: CC.Genesis.GeneratedSecrets
+    (genesisByron, generatedSecretsByron) =
+        Byron.generateGenesisConfig (toSlotLength byronSlotLenghtInSeconds) pbftParams
+
+    shelleyGenesis :: ShelleyGenesis c
+    shelleyGenesis =
+        Shelley.mkGenesisConfig
+          (hfSpecProtVer Shelley hardForkSpec)
+          securityParam
+          activeSlotCoeff
+          decentralizationParam
+          maxLovelaceSupply
+          (toSlotLength shelleySlotLengthInSeconds)
+          (Shelley.mkKesConfig (Proxy @c) numSlots)
+          [coreNodeShelley]
+      where
+        maxLovelaceSupply :: Word64
+        maxLovelaceSupply = 45000000000000000
+
+        activeSlotCoeff :: Rational
+        activeSlotCoeff = 0.2   -- c.f. mainnet is more conservative, using 0.05
+
+        numSlots = NumSlots 100
+
+-- | A more generalized version of 'mkSimpleTestProtocolInfo'.
+--
+mkTestProtocolInfo ::
+     forall m c
+   . (CardanoHardForkConstraints c, IOLike m)
+  => (CoreNodeId, Shelley.CoreNode c)
+  -- ^ Id of the node for which the protocol info will be elaborated.
+  -> ShelleyGenesis c
+  -- ^ These nodes will be part of the initial delegation mapping, and funds
+  --   will be allocated to these nodes.
+  -> CC.Update.ProtocolVersion
+  -- ^ Protocol version of the Byron era proposal.
+  -> SL.Nonce
+  -> CC.Genesis.Config
+  -> CC.Genesis.GeneratedSecrets
+  -> Maybe PBftSignatureThreshold
+  -> HardForkSpec
+  -- ^ Specification of the era to which the initial state should hard-fork to.
+  -> (ProtocolInfo (CardanoBlock c), m [BlockForging m (CardanoBlock c)])
+mkTestProtocolInfo
+    (coreNodeId, coreNode)
+    shelleyGenesis
+    aByronProtocolVersion
+    initialNonce
+    genesisByron
+    generatedSecretsByron
+    aByronPbftSignatureThreshold
+    hardForkSpec
+  =
+    protocolInfoCardano
+        ProtocolParamsByron {
+            byronGenesis                = genesisByron
+            -- Trivialize the PBFT signature window so that the forks induced by
+            -- the network partition are as deep as possible.
```

Comment:
```suggestion
```
This comment is already present here (could also be split across multiple lines there): https://github.com/input-output-hk/ouroboros-consensus/blob/c311626b6610f9efb079f5d49584070e65deee4a/ouroboros-consensus-cardano/test/cardano-test/Test/ThreadNet/Cardano.hs#L489